src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -60,11 +62,37 @@ def _mark_all_acquittals_ineligible_using_mrd_date(expunger):
 
     @staticmethod
     def _evaluate_class_b_felonies(expunger):
-        if expunger.most_recent_charge and expunger.most_recent_charge.disposition.date > TimeAnalyzer.TWENTY_YEARS_AGO:
-            for charge in expunger.class_b_felonies:
-                charge.set_time_ineligible('Time-ineligible under 137.225(5)(a)(A)(i)',
-                                           expunger.most_recent_charge.disposition.date + relativedelta(
-                                               years=TimeAnalyzer.TWENTY_YEARS))
+        for b_felony in expunger.class_b_felonies:
+
+            # if the B felony is acquitted, it is affected the other time eligibility rules,
+            # so leave it untouched here.
+            # if it's convicted, this restriction overrides any of those rules.
+            if not b_felony.acquitted():
+                has_subsequent_charge = False
+                for other_charge in expunger.charges:
```

Comment:
Doesn't `expunger.charges` include `expunger.class_b_felonies`? It looks like this fails the `date_of_charge > b_felony.disposition.date` check (as desired) because `date_of_charge == b_felony.disposition.date` when comparing the same charge, but it doesn't feel intentional.

A couple options include renaming `other_charge` to `charge`, removing `b_felony` from the charges like
`other_charges = [charge for charge in expunger.charges if charge != b_felony]`, or putting an explicit check for when `other_charge == b_felony`.

Reply:
Yes there is a self-comparison here, but a class B felony should only become ineligible if the compared date is strictly greater than that of the B felony. The self-equality here makes this expression false.  I suppose that explicitly pulling out the self-comparison could improve readability.

Reply:
Indeed, that is why I said "as desired".

Reply:
Ah, woops, yes good. I'm fine with changing it as a matter of style/readability.

Reply:
It was intentional though, slash I initially decided it wasn't worth acting on :)